resources/views/components/base/form/textarea.blade.php

```diff
@@ -79,18 +82,30 @@
                 <span class="ml-3">{{ $help }}</span>
             @endif
         </div>
-        <div>
-            @if($richText)
-                <x-fas-spinner id="preview-loading-icon" class="opacity-0 transition-all duration-200" aria-hidden="true" />
-                <button
-                    type="button"
-                    class="btn py-1 mt-1"
-                    onclick="window.loadPostPreview('{{ $id }}', 'post-preview-{{ $id }}')"
-                    :disabled="!isValid || isSending"
-                >{{ __('Preview') }}</button>
+
+        <div class="flex items-center gap-x-1">
+            <div>
+                @if ($richText)
+                    <x-fas-spinner
+                        id="preview-loading-icon"
+                        class="opacity-0 transition-all duration-200"
+                        aria-hidden="true"
+                    />
+
+                    <button
+                        type="button"
+                        class="btn btn-link"
+                        onclick="window.loadPostPreview('{{ $id }}', 'post-preview-{{ $id }}')"
+                        :disabled="!isValid || isSending"
+                    >
+                        {{ __('Preview') }}
+                    </button>
+                @endif
+            </div>
+
+            @if ($formActions)
+                {{ $formActions }}
             @endif
         </div>
     </div>
-
-    <div id="post-preview-{{ $id }}"></div>
```

Comment:
Moved this preview stuff out of the textarea component. I agree it's desirable to have it in one place, but not as part of the field or the form. This was causing some layout issues when the preview did appear in select areas of the UI.

Reply:
Couldn't this be just the default and be externalized when needed?
It's mainly about the id for me - being forced to assign ids to form elements manually isn't the best DX, i usually try to avoid that as it's error prone.

Reply:
Agreed, that sounds like a good compromise to me 👍 

These forum/message fields are unique in that their previews depend on being disjointed from the field (as they are themselves trying to simulate the layout of how the post will appear on the page). Other textareas probably won't need that unique requirement.

_textarea.blade.php_ updated with this recommendation in https://github.com/RetroAchievements/RAWeb/pull/2292/commits/d2da07c76d6fddc70f6f6ef532faae290f83589a.